pkg/controller/base/controller.go

```diff
@@ -38,16 +38,18 @@ type controller struct {
 	queue      workqueue.RateLimitingInterface
 	reconciler Reconciler
 
-	logger log.Logger
+	logger      log.Logger
+	threadiness int
 }
 
-func NewController(factories config.Factories, clients config.Clients, reconciler Reconciler, logger log.Logger) Controller {
+func NewController(threadiness int, factories config.Factories, clients config.Clients, reconciler Reconciler, logger log.Logger) Controller {
```

Comment:
Should this whole threadiness change be part of this PR?

Reply:
It an ideal world it should probably not. This is the first controller that requires this change though as `threadiness` doesn't apply for this controller. Instead of doing a separate PR only refactoring the controller interface I decided to shim it in here to avoid the roundtrips of 2 PRs. We can do it separately if you prefer.

Reply:
Ok